File: api-netcore/TasksReportManager/TasksReportManager.EntitiesModel/ActivityTask.cs
namespace TasksReportManager.EntitiesModel
{
  public class ActivityTask
    : EntityBase
  {

    public int TaskTypeId { get; set; } = 0;
    public TaskType TaskType { get; set; } = new TaskType();
    public int ActivityId { get; set; } = 0;
    public Activity Activity { get; set; } = new Activity();
    public string Url { get; set; } = string.Empty;
    public TimeSpan TimeElapsed { get; set; } = TimeSpan.FromSeconds(0);

  }
}

File: src/app/models/ActivityTask.ts
import { Activity } from "./Activity";
import { BaseModel } from "./BaseModel";
import { TaskType } from "./TaskType";

export class ActivityTask 
    extends BaseModel
{
    public taskTypeId: number = 0;
    public activityId: number = 0;
    public activity: Activity = new Activity();
    public taskType: TaskType = new TaskType();
    public url: string = "";
    public timeElapsed: string = "00:00";
    public timeElapsedEntry: string = "00:00";

    constructor() {
        super();
    }

    public static override toUrlResource(): string {
        return "ActivityTask";
    }
}